api/repositories/route_repository.go

```diff
@@ -175,7 +177,12 @@ func (r *RouteRepo) GetRoute(ctx context.Context, authInfo authorization.Info, r
 		return RouteRecord{}, fmt.Errorf("failed to get route %q: %w", routeGUID, apierrors.FromK8sError(err, RouteResourceType))
 	}
 
-	return cfRouteToRouteRecord(*route), nil
+	record, err := r.cfRouteToRouteRecord(ctx, authInfo, *route)
+	if err != nil {
```

Comment:
Not sure whether wrapping the conversion error brings any value. If not, we could just 

```
return r.cfRouteToRouteRecord(ctx, authInfo, *route)
```

That also applies to all error wrapping around the function invocations in create, delete, etc.